### PR TITLE
fix: Add back testing for all types

### PR DESCRIPTION
**Pull request by yevgenypats**
There is still more work to do to make the testing function "production" ready because they become quite vital and some of them are public APIs so we need to think how to generalize them correctly so they are testable, expand etc.. but this should be a good start just so we can start releasing some destinations

### CODE REVIEW

File: schema/testdata.go [hunk 1]
```diff
@@ -242,19 +64,102 @@ func columnsToFields(columns ...Column) []arrow.Field {
...
+	if !opts.SkipTimes {
+		columns = append(columns, ColumnList{
+			{Name: "time32_s", Type: arrow.FixedWidthTypes.Time32s},
+			{Name: "time32_ms", Type: arrow.FixedWidthTypes.Time32ms},
+			{Name: "time64_us", Type: arrow.FixedWidthTypes.Time64us},
+			{Name: "time64_ns", Type: arrow.FixedWidthTypes.Time64ns},
+		}...)
+	}
+
+	if !opts.SkipTimestamps {
```
Comment (candiduslynx) [inline schema/testdata.go:127]: Theres' an issue here that @hermanschaaf covered in original implementation: we added the `timestamp_us` option always, as this was the default transformed from old cq time.

File: plugin/testing_upsert.go [hunk 2]
```diff
@@ -63,3 +63,50 @@ func (s *WriterTestSuite) testUpsert(ctx context.Context) error {
...
+	tableName := fmt.Sprintf("cq_upsert_all_%d", time.Now().Unix())
+	table := schema.TestTable(tableName, s.genDatOptions)
+	table.Columns = append(table.Columns, schema.Column{Name: "name", Type: arrow.BinaryTypes.String, PrimaryKey: true})
+	if err := s.plugin.writeOne(ctx, &message.WriteMigrateTable{
+		Table: table,
+	}); err != nil {
+		return fmt.Errorf("failed to create table: %w", err)
+	}
+
+	records := schema.GenTestData(table, schema.GenTestDataOptions{MaxRows: 1})
```
Comment (hermanschaaf) [inline plugin/testing_upsert.go:77]: I think we should use `MaxRows: 2` here (and update the expectations accordingly) so that we are also testing a row with null values

File: plugin/testing_write_insert.go [hunk 3]
```diff
@@ -71,3 +71,53 @@ func (s *WriterTestSuite) testInsert(ctx context.Context) error {
...
+func (s *WriterTestSuite) testInsertAll(ctx context.Context) error {
+	tableName := fmt.Sprintf("cq_insert_all_%d", time.Now().Unix())
+	table := schema.TestTable(tableName, s.genDatOptions)
+	if err := s.plugin.writeOne(ctx, &message.WriteMigrateTable{
+		Table: table,
+	}); err != nil {
+		return fmt.Errorf("failed to create table: %w", err)
+	}
+	records := schema.GenTestData(table, schema.GenTestDataOptions{
+		MaxRows: 1,
```
Comment (hermanschaaf) [inline plugin/testing_write_insert.go:84]: Same comment here

File: plugin/testing_upsert.go (hunk 2 above)
Comment (yevgenypats) [reply plugin/testing_upsert.go]: I did `MaxRows=2` for now but I think it should be removed moving forward as it is a "control flag" which doesn't indicate what it does to the full extant. and we should just have an option `WithNulls` and then just call `GenTestData` twice

File: schema/testdata.go (hunk 1 above)
Comment (yevgenypats) [reply schema/testdata.go:127]: I'll address it in a follow-up PR.

File: plugin/testing_upsert.go (hunk 2 above)
Comment (hermanschaaf) [reply plugin/testing_upsert.go]: Agreed, that would be better 👍

File: schema/testdata.go (hunk 1 above)
Comment (hermanschaaf) [reply schema/testdata.go:127]: It's relatively easy to fix I think, just add `timestamp_us` to the initial list of columns

**Review by hermanschaaf — APPROVED** [review]
